test-suite/shared/src/test/require-jdk11/org/scalajs/testsuite/javalib/lang/MathTestOnJDK11.scala

```diff
@@ -0,0 +1,162 @@
+    test(-672138807810988440L, 531516745, -1264567512)
+    test(-193351903065245331L, 170858169, -1131651499)
+    test(71263251057597648L, 51058196, 1395725988)
+    test(-774312974742971385L, 1958551603, -395349795)
+    test(-1846593638370672048L, 1190143097, -1551572784)
+    test(240083094242536384L, 1404614968, 170924488)
+    test(-130950827889833280L, -115480554, 1133964320)
+    test(128954457719585228L, 735993884, 175211317)
+    test(364779990580792000L, -668489125, -545678272)
+    test(107252402494512045L, 759517757, 141211185)
+    test(3038084150893069044L, -1924640913, -1578519988)
+    test(760804294233336624L, -728394552, -1044494762)
+    test(1171051779605774913L, 848233701, 1380576813)
+    test(-1805862307837393080L, -1385644986, 1303264780)
+    test(172227703288618734L, -104999826, -1640266559)
+    test(150448013961014407L, 163398103, 920745169)
+    test(-671469201380991232L, 650262784, -1032612073)
+    test(-1325861126942924945L, -1773644581, 747534845)
+    test(987406376890116568L, -1626507773, -607071416)
+    test(2918138947401192144L, 1695881208, 1720721318)
+    test(-2590993826910153940L, -1397240042, 1854365570)
+    test(954644624447419276L, -1516139806, -629654746)
+    test(407510452326678620L, -384747652, -1059162935)
+    test(149866317537821404L, 1530355444, 97929091)
+    test(922044716091910632L, 968149268, 952378674)
+    test(-3508732521573808284L, 1825364562, -1922209182)
+    test(1701723136959404304L, 894776752, 1901841027)
+    test(-2435876799625512705L, -1276062909, 1908900245)
+    test(-516933170985379201L, 657063047, -786732983)
+    test(123334479976750576L, 313765817, 393078128)
+    test(-1072624004420456775L, -894199299, 1199535725)
+    test(301682711612188737L, 330918981, 911651277)
+    test(1790992996470651507L, -1115945231, -1604911197)
+    test(-2750453268538140155L, 1878389719, -1464261245)
+    test(758285757353272504L, 1259684942, 601964612)
+    test(-218581674312137400L, -161533394, 1353167100)
+    test(-1824007072461951836L, -1244277844, 1465916219)
+    test(-92753167730460334L, -65368843, 1418920138)
+    test(-2326636630979491248L, 1124395877, -2069232624)
+    test(-7380586257943446L, 29715454, -248375349)
+    test(31319707234597638L, 491995506, 63658523)
+    test(-1196559502630778250L, -1752963990, 682592175)
+    test(166065559841839548L, -911521074, -182185102)
+    test(-1222260378510810100L, 1071539812, -1140657925)
+    test(57800571165871464L, -257569032, -224408077)
+    test(332444627169725608L, 1247224172, 266547614)
+    test(217903869180130650L, 1069161915, 203808110)
+    test(920425054266935850L, -901689546, -1020778225)
+    test(-507632632656614388L, 864632142, -587108214)
+  }
+
+  @Test def testMultiplyHigh(): Unit = {
+    /* We fuzz-test by comparing to the "obvious" implementations based on
+     * BigIntegers. We use a SplittableRandom generator, because Random cannot
+     * generate all Long values.
+     */
```

Comment:
Doing this in runtime concerns me, because it relies on our own BigInteger implementation for correctness (in fact, nothing seems to prevent our BigInteger implementation to rely on the math implementation here that we are testing).

Do you think we could just expand some test cases out? (Given the complexity of the method under test, I think 50 or 100 cases like above should be way enough).

Reply:
Fair enough. I'll expand the test cases.